packages/entity-cache-adapter-redis/src/GenericRedisCacher.ts

```diff
@@ -44,8 +44,39 @@ export enum RedisCacheInvalidationStrategy {
    * This strategy generates cache keys for both old and potential future new versions.
    */
   SURROUNDING_CACHE_KEY_VERSIONS = 'surrounding-cache-key-versions',
+
+  /**
+   * Invalidate cache keys based on user-specified function from the current cacheKeyVersion to a list of cache key
+   * versions to invalidate.
+   */
+  CUSTOM = 'custom',
 }
 
+export type GenericRedisCacheInvalidationConfig =
+  | {
+      /**
+       * Invalidation strategy for the cache.
+       */
+      invalidationStrategy: RedisCacheInvalidationStrategy.CURRENT_CACHE_KEY_VERSION;
+    }
+  | {
+      /**
+       * Invalidation strategy for the cache.
+       */
+      invalidationStrategy: RedisCacheInvalidationStrategy.SURROUNDING_CACHE_KEY_VERSIONS;
+    }
+  | {
+      /**
+       * Invalidation strategy for the cache.
+       */
+      invalidationStrategy: RedisCacheInvalidationStrategy.CUSTOM;
```

Comment:
would be nice to describe the strategies in the docblock

Reply:
Yep, they're detailed in the enum docblock! These docblocks all are the same since it's a discriminated union and most editors just take an arbitrary one anyways.